rosbag2_compression/CMakeLists.txt

```diff
@@ -45,6 +45,12 @@ ament_target_dependencies(${PROJECT_NAME}
   rosbag2_storage)
 target_compile_definitions(${PROJECT_NAME} PRIVATE ROSBAG2_COMPRESSION_BUILDING_DLL)
 
+if(CMAKE_COMPILER_IS_GNUCXX)
+  target_link_libraries(${PROJECT_NAME}
+    rcpputils::filesystem
+  )
+endif()
+
```

Comment:
@audrow @clalancette I think we need 
```
if(CMAKE_COMPILER_IS_GNUCXX)
  target_link_libraries(${PROJECT_NAME} stdc++fs)
#endif
```
IMO ` target_link_libraries(${PROJECT_NAME} rcpputils::filesystem)` is not going to help here.

Reply:
This actually isn't needed, as a similar thing is done above with `ament_target_dependencies`.

Instead, we need a change to the code itself to switch back to using `rcpputils::fs`.  I'll push some changes shortly that make it compile for me, though @MichaelOrlov suggests that it may not work in practice.

Reply:
@clalancette Yes, it is not going to work out of the box. I will do rework shortly. Give me a couple of hours.